benchmark/infrastructure/BenchmarkResultToString.cpp

```diff
@@ -4,55 +4,196 @@
+                          const size_t& indentionLevel) {
+  // An indention level of 0 makes no sense. Must be an error.
+  AD_CONTRACT_CHECK(indentionLevel > 0);
+
+  // The indention symbols for this level of indention.
+  std::string indentionSymbols{""};
+  indentionSymbols.reserve(outputIndention.size() * indentionLevel);
+  for (size_t i = 0; i < indentionLevel; i++) {
+    indentionSymbols.append(outputIndention);
+  }
+
+  std::ostringstream stream;
+  stream << indentionSymbols;
+
+  // Adding the content of the string char per char, so that we can easily
+  // identify the new line symbols.
+  std::ranges::for_each(str,
+                        [&indentionSymbols, &stream](const char& symbol) {
+                          stream << symbol;
+                          if (symbol == '\n') {
+                            stream << indentionSymbols;
+                          }
+                        },
+                        {});
+
+  return stream.str();
+}
+
+// ___________________________________________________________________________
+std::string getMetadataPrettyString(const BenchmarkMetadata& meta,
+                                    std::string_view prefix,
+                                    std::string_view suffix) {
+  const std::string& metadataString = meta.asJsonString(true);
+  if (metadataString != "null") {
+    return absl::StrCat(prefix, metadataString, suffix);
+  } else {
+    return "";
+  }
+}
+
+/*
+@brief Applies the given function `regularFunction` to all elements in `r`,
+except for the last one. Instead, `lastOneFunction` is applied to that one.
+*/
+template <typename Range, typename RegularFunction, typename LastOneFunction>
+static void forEachExcludingTheLastOne(Range&& r,
+                                       RegularFunction regularFunction,
+                                       LastOneFunction lastOneFunction) {
+  std::ranges::for_each_n(r.begin(), r.size() - 1, regularFunction, {});
+  lastOneFunction(r.back());
+}
+
+/*
+@brief Adds the elements of the given range to the stream in form of a list.
+
+@tparam TranslationFunction Must take the elements of the range and return a
+string.
+
+@param translationFunction Converts range elements into string.
+@param listItemSeperator Will be put between each of the string representations
+of the range elements.
+*/
+template <typename Range, typename TranslationFunction>
+static void addListToOStringStream(std::ostringstream* stream, Range&& r,
+                                   TranslationFunction translationFunction,
+                                   std::string_view listItemSeperator = "\n") {
+  forEachExcludingTheLastOne(
+      AD_FWD(r),
+      [&stream, &translationFunction,
+       &listItemSeperator](const auto& listItem) {
+        (*stream) << translationFunction(listItem) << listItemSeperator;
+      },
+      [&stream, &translationFunction](const auto& listItem) {
+        (*stream) << translationFunction(listItem);
+      });
```

Comment:
Add a `TODO<C++23>: This can be a combination of `std::views::transform` and `std::views::join_with`.

Reply:
Done, see [3b5da81](https://github.com/ad-freiburg/qlever/pull/958/commits/3b5da8183e347249179788e3efff0820584f39a3).